pkg/ccl/backupccl/backuputils/testutils.go

```diff
@@ -114,3 +118,60 @@ func BackupRestoreTestSetup(
 ) (tc *testcluster.TestCluster, sqlDB *sqlutils.SQLRunner, tempDir string, cleanup func()) {
 	return backupRestoreTestSetupWithParams(t, clusterSize, numAccounts, init, base.TestClusterArgs{})
 }
+
+// VerifyBackupRestoreStatementResult conducts a Backup or Restore and verifies
+// it was properly written to the jobs table
+func VerifyBackupRestoreStatementResult(
+	t *testing.T, sqlDB *sqlutils.SQLRunner, query string, args ...interface{},
+) error {
+	t.Helper()
+	rows := sqlDB.Query(t, query, args...)
+
+	columns, err := rows.Columns()
+	if err != nil {
+		return err
+	}
+	if e, a := columns, []string{
+		"job_id", "status", "fraction_completed", "rows", "index_entries", "bytes",
+	}; !reflect.DeepEqual(e, a) {
+		return errors.Errorf("unexpected columns:\n%s", strings.Join(pretty.Diff(e, a), "\n"))
+	}
+
+	type job struct {
+		id                int64
+		status            string
+		fractionCompleted float32
+	}
+
+	var expectedJob job
+	var actualJob job
+	var unused int64
+
+	if !rows.Next() {
+		if err := rows.Err(); err != nil {
+			return err
+		}
+		return errors.New("zero rows in result")
+	}
+	if err := rows.Scan(
+		&actualJob.id, &actualJob.status, &actualJob.fractionCompleted, &unused, &unused, &unused,
+	); err != nil {
+		return err
+	}
+	if rows.Next() {
+		return errors.New("more than one row in result")
+	}
+
+	sqlDB.QueryRow(t,
+		`SELECT job_id, status, fraction_completed FROM crdb_internal.jobs WHERE job_id = $1`, actualJob.id,
+	).Scan(
+		&expectedJob.id, &expectedJob.status, &expectedJob.fractionCompleted,
+	)
+
+	if e, a := expectedJob, actualJob; !reflect.DeepEqual(e, a) {
+		return errors.Errorf("result does not match system.jobs:\n%s",
+			strings.Join(pretty.Diff(e, a), "\n"))
+	}
+
```

Comment:
If you felt like it, I think `require.Equal` provides pretty nice output by default and could simplify this function a bit.

Reply:
Done.